psyneulink/core/compositions/composition.py

```diff
@@ -9631,7 +9796,12 @@ def execute(
                             context=context
                         )
                     ):
-                        node.reset(None, context=context)
+                        if node in reset_stateful_functions_to:
+                            node.reset(*reset_stateful_functions_to[node],
+                                       context=context)
+                        else:
+                            node.reset(None,
+                                       context=context)
```

Comment:
```
vals = reset_state_functions_to.get(node, [None])
node.reset(*vals, context=context)
```
would save one lookup in the successful case

Reply:
updated in ffad3efb0747fedd8306bad83a2bfc5dd5ddee29